src/main/java/org/apache/maven/shared/archiver/PomPropertiesUtil.java

```diff
@@ -61,36 +60,51 @@ private boolean sameContents(Properties props, Path file) throws IOException {
         return fileProps.equals(props);
     }
 
-    private void createPropertiesFile(Properties properties, Path outputFile, boolean forceCreation)
+    private void createPropertiesFile(Properties unsortedProperties, Path outputFile, boolean forceCreation)
             throws IOException {
         Path outputDir = outputFile.getParent();
         if (outputDir != null && !Files.isDirectory(outputDir)) {
             Files.createDirectories(outputDir);
         }
-        if (!forceCreation && sameContents(properties, outputFile)) {
+        if (!forceCreation && sameContents(unsortedProperties, outputFile)) {
             return;
         }
 
-        try (PrintWriter pw = new PrintWriter(outputFile.toFile(), StandardCharsets.ISO_8859_1.name());
-                StringWriter sw = new StringWriter()) {
-
-            properties.store(sw, null);
-
-            List<String> lines = new ArrayList<>();
-            try (BufferedReader r = new BufferedReader(new StringReader(sw.toString()))) {
-                String line;
-                while ((line = r.readLine()) != null) {
-                    if (!line.startsWith("#")) {
-                        lines.add(line);
-                    }
-                }
+        // For reproducible builds, sort the properties and drop comments.
+        // The java.util.Properties class doesn't guarantee order so we have
+        // to write the file using a Writer.
+        Set<String> propertyNames = unsortedProperties.stringPropertyNames();
+        List<String> sortedPropertyNames = new ArrayList<>(propertyNames);
+        Collections.sort(sortedPropertyNames);
+
+        try (Writer out = Files.newBufferedWriter(outputFile, StandardCharsets.ISO_8859_1)) {
+            for (String key : sortedPropertyNames) {
+                out.write(escape(key));
+                out.write("=");
+                out.write(escape(unsortedProperties.getProperty(key)));
+                out.write('\n');
             }
+        }
+    }
 
-            Collections.sort(lines);
-            for (String l : lines) {
-                pw.println(l);
+    private static String escape(String s) {
+        StringBuilder sb = new StringBuilder(s.length());
+        for (char c : s.toCharArray()) {
+            if (Character.isWhitespace(c) || c == '#' || c == '!' || c == '=' || c == ':') { // backslash escape
+                sb.append('\\');
+                sb.append(c);
+            } else if (c < 256) { // 8859-1
```

Comment:
The test is wrong.

Reply:
OK,. this one I don't see. What's wrong here?

Reply:
The following test succeeds:
```
    Properties p = new Properties();
        p.put("foo", "aéüb");
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        p.store(baos, null);
        String s = baos.toString();
        assertTrue(s.contains("foo=a\\u00E9\\u00FCb"));
```
So non plain ascii chars should be unicode encoded.